fcrepo-kernel-api/src/main/java/org/fcrepo/kernel/api/models/ResourceFactory.java

```diff
@@ -112,4 +112,17 @@ public FedoraResource getResource(final String transactionId, final FedoraId fed
      * @return Stream of child resources
      */
     public Stream<FedoraResource> getChildren(final String transactionId, final FedoraId resourceId);
+
+    /**
+     * Is the resource a "ghost node". Ghost nodes are defined as a resource that does not exist, but whose URI is part
+     * of the URI of another resource? For example:
+     *
+     * http://localhost/rest/a/b - does exist
+     * http://localhost/rest/a - does not exist and is therefore a ghost node.
+     *
+     * @param transaction The transaction
+     * @param resourceId Identifier of the resource
+     * @return Whether the resource does not exist, but has
+     */
+    public boolean isGhostNode(final Transaction transaction, final FedoraId resourceId);
```

Comment:
This seems more like a method that would be on `FedoraResource` than `ResourceFactory`.

..such as `isAcl()` and `isMemento()`.

Reply:
The use of `isGhostNode` in the mintNewPid, seems to make adding this method to `FedoraResource` make less sense. Alternatively I can move the actual check down to the `AbstractService` which also has access to the `ContainmentIndex`.

Reply:
Actually the use of `isGhostNode` in the `mintNewPid` means this needs to exist here, because in that case we don't fail we just mint a new pid. So I think the current location is best and the use of `ResourceFactory` is just because we don't Inject the `ContainmentIndex` to the Http layer.